source/common/singleton/manager_impl.cc

```diff
@@ -0,0 +1,26 @@
+#include "common/singleton/manager_impl.h"
+
+#include "envoy/registry/registry.h"
+
+namespace Envoy {
+namespace Singleton {
+
+InstancePtr ManagerImpl::get(const std::string& name) {
+  verifyRegistration(name);
```

Comment:
Does it make sense to have any thread checks here? Can anyone other than the main thread access this? IF so, do we need to lock to avoid racing on get/set?